### PR TITLE
Mark all buffers errored if server times out

### PR DESCRIPTION
**Pull request by ptrv**

### CODE REVIEW

File: ycmd.el [hunk 1]
```diff
@@ -629,9 +634,8 @@ This does nothing if no server is running."
   (ycmd--kill-timer ycmd--keepalive-timer))
 
 (defun ycmd-running? ()
-  "Tells you if a ycmd server is already running."
-  (interactive)
```
Comment (ptrv) [inline ycmd.el]: Does not need to be interactive

File: ycmd.el [hunk 2]
```diff
@@ -1320,13 +1324,14 @@ the name of the newly created file."
...
-                (setq cont nil)))
+                (setq cont nil)
+                (ycmd--do-ycmd-buffers (ycmd--report-status 'unparsed))))
              (t
               ;; timeout after specified period
               (when (< (/ ycmd-startup-timeout 1000) (- (float-time) start-time))
-                (with-current-buffer orig-buff
-                  (ycmd--report-status 'errored))
-                (error "Server timeout"))))))))))
+                (ycmd--do-ycmd-buffers (ycmd--report-status 'errored))
```
Comment (ptrv) [inline ycmd.el]: Close server if we have a process but cannot parse the output.

File: ycmd.el [hunk 3]
```diff
@@ -567,6 +567,14 @@ _LEN is ununsed."
     (setq ycmd--on-focus-timer
           (run-at-time 1.0 nil #'ycmd--on-unparsed-buffer-focus))))
 
+(defmacro ycmd--with-all-ycmd-buffers (&rest body)
+  "Execute BODY with each `ycmd-mode' enabled buffer."
+  (declare (indent 0) (debug t))
```
Comment (abingham) [inline ycmd.el]: Should this line be removed?

**Comment by abingham** [comment]
Besides the one question I had, this looks good.

File: ycmd.el (hunk 3 above)
Comment (ptrv) [inline ycmd.el]: It shouldn't be removed. The `declare` form is a special form to add meta data to macros or functions.
See `(info "(elisp) Declare Form")`

Comment (abingham) [inline ycmd.el]: OK, cool, I just wasn't sure if it was development scaffolding. Learning more each day!

**Pull request merged**